identity-tests/src/main/scala/com/gu/identity/integration/test/pages/FaceBookAuthDialog.scala

```diff
@@ -0,0 +1,43 @@
+package com.gu.identity.integration.test.pages
+
+import com.gu.integration.test.pages.common.ParentPage
+import org.openqa.selenium.support.ui.{WebDriverWait, ExpectedConditions}
+import org.openqa.selenium.{By, WebDriver, WebElement}
+
+class FaceBookAuthDialog(implicit driver: WebDriver) extends ParentPage {
+  private def confirmButton: WebElement = {
+    val xpath = "id('platformDialogForm')//button[@name='__CONFIRM__']"
```

Comment:
Agreed, it's not ideal. It basically boils down to two things:

1) The `login_url` that Facebook returns as part of the Test User API is broken. It is supposed to log in the user automatically, but it is no longer working (http://stackoverflow.com/questions/22583963/404-error-trying-to-use-facebook-test-user-login-url/22751523#22751523). That is why the login uses the regular Facebook login, and since this is part of the OAuth flow for a Facebook application, the test user will be presented with that dialog.

2) There is an option to create the test user with the Facebook application already installed. While that would simplify things, one of the things we're testing is "what if the user does not allow e-mail permissions?". We could revoke the e-mail permission specifically via the Graph API, but that would have to be done outside of the OAuth flow, so the tests wouldn't really be representative of what a user would be doing (which is interacting with the dialog box).

I'll take a look at Koala and see if they're doing something that might help make the tests a bit more sustainable.

Thanks.